physics/closed_system.hpp

```diff
@@ -0,0 +1,86 @@
+﻿#pragma once
+
+#include <optional>
+#include <vector>
+
+#include "physics/inertia_tensor.hpp"
+#include "physics/rigid_motion.hpp"
+
+namespace principia {
+namespace physics {
+namespace internal_closed_system {
+
+using geometry::BarycentreCalculator;
+using geometry::Bivector;
+using geometry::Frame;
+using geometry::OrthogonalMap;
+using geometry::SymmetricBilinearForm;
+using quantities::AngularMomentum;
+using quantities::Mass;
+using quantities::MomentOfInertia;
+
+// Computes the instantaneous overall properties of a mechanical system.
+// Effectively an extension of
+// |BarycentreCalculator<DegreesOfFreedom<InertialFrame>, Mass>| that also
+// handles rotational motion.
+// Note: As a consequence of our separate handling of vectors and bivectors, the
+// inertia tensor used here does *not* follow the convention customary in
+// physics.  The usual convention is
+//  ⎛ ∑(y² + z²)   -∑xy      -∑xy    ⎞
+//  ⎜    -∑yx   ∑(x² + z²)   -∑yz    ⎟
+//  ⎝    -∑zx      -∑zy   ∑(x² + y²) ⎠
+// but our convention is:
+//  ⎛ ∑x²   ∑xy   ∑xy ⎞
+//  ⎜ ∑yx   ∑y²   ∑yz ⎟
+//  ⎝ ∑zx   ∑zy   ∑z² ⎠
+// This leads to simpler transformation formulæ.  Where the physics usage would
+// be to multiply the inertia tensor with a pseudovector, we must instead take
+// the anticommutator with the bivector: L = Iω becomes L = {I, ω}, and the
+// moment of inertia αIα along an axis α becomes ⟨α, {I, α}⟩.
+template<typename InertialFrame, typename SystemFrame>
```

Comment:
1. Add comments to explain the purpose of the frames.
1. Should there be an assert that `InertialFrame` is inertial?